From linux-kernel-owner+greg=40kroah.com-S965240AbVIOHcf@vger.kernel.org Thu Sep 15 00:47:08 2005
Message-Id: <20050915070302.574089000.dtor_core@ameritech.net>
Date: Thu, 15 Sep 2005 02:01:37 -0500
From: Dmitry Torokhov <dtor_core@ameritech.net>
Cc: Andrew Morton <akpm@osdl.org>, Greg KH <gregkh@suse.de>, Kay Sievers <kay.sievers@vrfy.org>, Vojtech Pavlik <vojtech@suse.cz>, Hannes Reinecke <hare@suse.de>
Subject: Driver core: send hotplug event before adding class interfaces
Patch-mainline: 2.6.15-rc1

Move call to kobject_hotplug() above code that adds interfaces
to a class device, otherwise children's hotplug events may reach
userspace first.

Signed-off-by: Dmitry Torokhov <dtor@mail.ru>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

 drivers/base/class.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletion(-)

--- linux-2.6.13.orig/drivers/base/class.c	2005-10-27 09:59:17.000000000 -0700
+++ linux-2.6.13/drivers/base/class.c	2005-10-27 09:59:20.000000000 -0700
@@ -535,6 +535,8 @@
 				  class_name);
 	}
 
+	kobject_hotplug(&class_dev->kobj, KOBJ_ADD);
+
 	/* notify any interfaces this device is now here */
 	if (parent) {
 		down(&parent->sem);
@@ -544,7 +546,6 @@
 				class_intf->add(class_dev);
 		up(&parent->sem);
 	}
-	kobject_hotplug(&class_dev->kobj, KOBJ_ADD);
 
  register_done:
 	if (error && parent)
